pkg/cloud/services/ec2/ami.go

```diff
@@ -94,8 +103,46 @@ func GenerateAmiName(amiNameFormat, baseOS, kubernetesVersion string) (string, e
 	return templateBytes.String(), nil
 }
 
+// Determine architecture based on instance type.
+func (s *Service) pickArchitectureForInstanceType(instanceType string) (string, error) {
+	descInstanceTypeInput := &ec2.DescribeInstanceTypesInput{
+		InstanceTypes: []*string{&instanceType},
+	}
+	describeInstanceTypeResult, err := s.EC2Client.DescribeInstanceTypes(descInstanceTypeInput)
+	if err != nil {
+		return "", err
+	}
+
+	supportedArchs := describeInstanceTypeResult.InstanceTypes[0].ProcessorInfo.SupportedArchitectures
+
+	logger := s.scope.GetLogger().WithValues("instance type", instanceType, "supported architectures", supportedArchs)
+	logger.Info("Obtained a list of supported architectures for instance type")
+
+	// Loop over every supported architecture for the instance type
+	architecture := ""
+archCheck:
+	for _, a := range supportedArchs {
```

Comment:
Okay so sorry but I'm still struggling to understand how this switch works. 🤣🤣 so in supported architecture if the first item is amd we will just accept that but if the first item is arn then we will just accept THAT! Like how is this determinate in any way or how is this a "choice"? Sorry I just might not know the api here very well.

Reply:
Yeah, this is what I was expecting. However... This means that if both arm and amd are available, there is no choice, right? It will return whichever is first.

Reply:
Okay then. :)

Reply:
In that case, please just add the len check, than it's good to go. :)